packages/openapi3/src/openapi.ts

```diff
@@ -218,42 +219,47 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
 
   async function emitOpenAPI() {
     try {
-      for (let route of getRoutes(program)) {
-        if (route.kind !== "Namespace" && route.kind !== "Interface") {
-          reportDiagnostic(program, {
-            code: "resource-namespace",
-            target: route,
-          });
-          continue;
-        }
+      const routes = getRoutes(program);
+      const versions = getVersions(program, routes[0]) ?? [undefined];
+      for (const [index, version] of versions.entries()) {
+        currentVersion = version;
+        for (let route of routes) {
+          if (route.kind !== "Namespace" && route.kind !== "Interface") {
+            reportDiagnostic(program, {
+              code: "resource-namespace",
+              target: route,
+            });
+            continue;
+          }
 
-        // An interface can have @route applied but is handled by emitInterface
-        if (route.kind === "Namespace") {
-          emitRoute(route as NamespaceType);
+          // An interface can have @route applied but is handled by emitInterface
+          if (route.kind === "Namespace") {
+            emitRoute(route as NamespaceType);
+          }
         }
-      }
 
-      const actualNamespace = getServiceNamespace(program);
-      if (actualNamespace) {
-        emitInterfaces(actualNamespace);
-      }
+        const actualNamespace = getServiceNamespace(program);
+        if (actualNamespace) {
+          emitInterfaces(actualNamespace);
+        }
 
-      emitReferences();
-      emitTags();
+        emitReferences();
+        emitTags();
 
-      // Clean up empty entries
-      for (let elem of Object.keys(root.components)) {
-        if (Object.keys(root.components[elem]).length === 0) {
-          delete root.components[elem];
+        // Clean up empty entries
+        for (let elem of Object.keys(root.components)) {
+          if (Object.keys(root.components[elem]).length === 0) {
+            delete root.components[elem];
+          }
         }
-      }
 
-      if (!program.compilerOptions.noEmit && !program.hasError()) {
-        // Write out the OpenAPI document to the output path
-        await program.host.writeFile(
-          path.resolve(options.outputFile),
-          prettierOutput(JSON.stringify(root, null, 2))
-        );
+        if (!program.compilerOptions.noEmit && !program.hasError()) {
+          // Write out the OpenAPI document to the output path
+          await program.host.writeFile(
+            path.resolve(options.outputFile.replace(".json", `.v${index}.json`)),
```

Comment:
wouldn't that be better to be the version instead of the index

Reply:
There's nothing that says a version string has to be a valid file path but yeah probably anyway!